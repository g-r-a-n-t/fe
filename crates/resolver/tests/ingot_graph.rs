use std::str::FromStr;

use camino::Utf8PathBuf;
use fe_resolver::{ingot::basic_ingot_graph_resolver, Resolver};
use test_utils::snap_test;

#[test]
fn ingot_graph_resolution() {
    let mut resolver = basic_ingot_graph_resolver();
    let ingot_graph = resolver
        .transient_resolve(
            &Utf8PathBuf::from_str("fixtures/ingots/A")
                .unwrap()
                .canonicalize_utf8()
                .unwrap(),
        )
        .unwrap();
    snap_test!(
        format!("{:#?}\n{:#?}", ingot_graph, resolver.take_diagnostics()),
        "../../../fixtures/ingots"
    );
}
