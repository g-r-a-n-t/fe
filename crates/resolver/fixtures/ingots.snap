---
source: crates/resolver/tests/ingot_graph.rs
expression: "format!(\"{:#?}\\n{:#?}\", ingot_graph, resolver.take_diagnostics())"
---
Graph {
    Ty: "Directed",
    node_count: 5,
    edge_count: 6,
    edges: (0, 1), (1, 2), (1, 3), (3, 0), (0, 4), (4, 1),
    node weights: {
        0: "/home/grantwuerker/workshop/fe/crates/resolver/fixtures/ingots/A",
        1: "/home/grantwuerker/workshop/fe/crates/resolver/fixtures/ingots/C",
        2: "/home/grantwuerker/workshop/fe/crates/resolver/fixtures/ingots/E",
        3: "/home/grantwuerker/workshop/fe/crates/resolver/fixtures/ingots/D",
        4: "/home/grantwuerker/workshop/fe/crates/resolver/fixtures/ingots/B",
    },
}
[]
